jsonapi/data_structs.go

```diff
@@ -0,0 +1,35 @@
+package jsonapi
+
+type document struct {
+	Links    links  `json:"links,omitempty"`
+	Data     data   `json:"data"`
+	Included []data `json:"included,omitempty"`
+}
+
+type links struct {
+	Self     string `json:"self,omitempty"`
+	Related  string `json:"related,omitempty"`
+	First    string `json:"first,omitempty"`
+	Previous string `json:"previous,omitempty"`
+	Next     string `json:"next,omitempty"`
+	Last     string `json:"last,omitempty"`
+}
+
+type data struct {
+	Type          string                  `json:"type"`
+	ID            string                  `json:"id"`
+	Attributes    map[string]interface{}  `json:"attributes"`
+	Relationships map[string]relationship `json:"relationships,omitempty"`
+	Links         links                   `json:"links,omitempty"`
+}
+
+type relationship struct {
+	Links      links              `json:"links,omitempty"`
+	DataToOne  relationshipData   `json:"data,omitempty"`
```

Comment:
don't know if this works here, but jsonapi does an array for to-many and an object for to-one